pkg/util/machine/kcp.go

```diff
@@ -19,17 +19,34 @@ package machine
 import (
 	goctx "context"
 
+	"github.com/pkg/errors"
 	apitypes "k8s.io/apimachinery/pkg/types"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	labelutil "github.com/smartxworks/cluster-api-provider-elf/pkg/util/labels"
 )
 
+// GetKCPNameByMachine returns the KCP name associated with the Machine.
+// Can not use "cluster.x-k8s.io/control-plane-name" label because
+// it's value will be a hash string when the KCP name exceeds 63 characters.
```

Comment:
it's value will be a hash string ->
its value will be a hashed string of the KCP name